app/controllers/passwords_controller.rb

```diff
@@ -18,10 +18,10 @@ def create
   end
 
   def update
-    user = User.find_by(reset_token: params[:token])
+    user = User.find_by(token: params[:token])
     user.password = params[:password]
-    user.reset_token = nil
-    user.reset_token_expiration = nil
+    user.token = nil
+    user.token_expiration = nil
     user.save!
```

Comment:
This approach has 1 flaw and thats when password would be invalid. Rather than `save!` I would recommend to do it differently. Also you could use `update!` method or `update` rather.

```ruby
if user&.update(password: params[:password])
  user.remove_token!
  user.remove_expiration_date
  # flash + redirect
elsif user
  # render back pw error
else
  # redirect
end
```